firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt

```diff
@@ -70,7 +70,12 @@ private abstract class FirestoreKtxAbstractDecoder(
         // TODO: Add a EnumNamingProperties parameter, and convert decodedEnumFieldName based on it
         // i.e. case insensitive, snake_case match camelCase, etc
         val enumFieldNames = enumDescriptor.elementNames.toList()
```

Comment:
You could do

```
val index = enumDescriptor.elementNames.indexOf(decodedEnumFieldName)
        if (index == -1)
            throw IllegalArgumentException(
                "Could not find a match for enum field name of $decodedEnumFieldName."
            )
        return index
```

that way you don't create a list unnecessarily

Reply:
I didn't realize that Kotlin has `indexOf` buildin for iterables!!! This is so cool. 
 I updated this method, and added a unit test for the throw message. Thank you.